frontend/src/components/ManageAdminCard.jsx

```diff
@@ -0,0 +1,153 @@
+import {PasswordInput, Card, Button, Stack, Text, Title, Flex, Space} from "@mantine/core";
+import {modals} from "@mantine/modals"
+import {useDisclosure } from "@mantine/hooks";
+import {isNotEmpty, useForm} from "@mantine/form";
+import {HashLink} from "react-router-hash-link";
+import axios from "axios";
+import { notifications } from "@mantine/notifications";
+
+// TO-DO !! prohibit users from deleting themselves!!
+
+export function ManageAdminCard({admin, h, w, light=false, showDate=true, imgHPct=0.6, ...others}) {
+    const { opened, toggle } = useDisclosure();
+    const { user } = admin;
+
+    const form = useForm({
+        mode: "uncontrolled",
+
+        initialValues: {
+            user: '',
+            oldPassword: '',
+            newPassword: '',
+            passwordConfirm: ''
+        },
+        
+        validate: {
+            oldPassword: isNotEmpty('Informe a velha senha'),
+            newPassword: isNotEmpty('Informe a nova senha'),
+            passwordConfirm: (value, values) => {
+                if (value !== values.newPassword) {
+                    return 'As senhas não coincidem!';
+                }
+                return isNotEmpty('Confirme a nova senha')(value);
+            }
+        }
+    });
+
+    let textH = h - 20;
+    const lineCount = Math.floor(textH / 32);
+
+    const bgColor = light ? "aprai-purple.3" : "aprai-purple.9";
+    const textColor = light ? "aprai-purple.9" : "white";
+
+    function onClick(){
+        const validationResults = form.validate();
+    
+        if (validationResults.hasErrors) {
+            const firstError = Object.values(validationResults.errors)[0]
+            notifications.show({ message: firstError, color: 'red' });
+            return;
+        }
+
+        const oldPassword = form.getInputProps('oldPassword').defaultValue
+        const newPassword = form.getInputProps('newPassword').defaultValue
+        const passwordConfirm = form.getInputProps('passwordConfirm').defaultValue
+
+        console.log(`user: ${user} past:${oldPassword} new:${newPassword} confirm:${passwordConfirm}`)
+        axios.put(`api/admins/${user}`, {
+            user: user,
+            oldPassword: oldPassword,
+            newPassword: newPassword
+        }).then(_ => {
+            notifications.show({message: 'Informação atualizada com sucesso.'})
+        }).catch(err => {
+            if (err.response.data.validationErrors){
```

Comment:
Maybe should also handle the error code 401 (invalid old password) instead of printing an error to the console? A custom message like in the login page? Or maybe setting form.errors.oldPassword.

Reply:
done (custom message)